fecgov/fec-cms
# Feature/3651 flake8 linting

## Summary

- Resolves #3651 
- Adds flake8 linting to `pytest` as mentioned in 

## Impacted areas of the application
- Added flake8 and updated the config files accordingly
- Copied pytest flake8 rules to the `.flake8` config file that IDEs use
- Exempted the `.vscode/settings.json` from git so my/local settings don't override others' preferences
- Lots of linting fixes that were simple formatting (e.g., too many or too few blank lines, too few/many spaces)
- Several linting changes that are less clear to me—for those I added `# noqa ____` lines, exempting that one rule for that one line

## Screenshots

Should be _zero_ front-end differences

## Related PRs

List related PRs against other branches:

branch | PR
------ | ------
feature/4258-add-flake8 | [Open FEC #4259](https://github.com/fecgov/openFEC/pull/4259)

## How to test

- Pull the branch
- `pip install -r requirements.txt`
- `npm i` (just in case? Just a habit for me)
- `npm run build` (another habit)
- Review file changes
  - Check **all** files with `# noqa` in them. Ideally every instance should be fixed and the noqa exception removed. There were a couple I removed that broke things so I'd like those more familiar with that code to double-check
  - Check indentation? I think they're all correct but there was one I had to adjust a couple times. Would be nice to have other eyes on that type of thing
- `pytest` should run all of the tests
- `pytest --linting` should run only the Python linting
- All tests should pass

## Next steps
To me, it feels like the python linting should also happen inside `npm run watch` and `npm run build-production` but I'm still working through that. Not a required part of this ticket

____
